### PR TITLE
[MISC] split io/stream/iterator.hpp into multiple header

### PR DESCRIPTION
**Pull request by marehr**

### CODE REVIEW

**Review by joergi-w — CHANGES_REQUESTED** [review]
Thank you! I would like you to answer two questions and make one change:
1. Why don't we deprecate the empty file `iterator.hpp`? It is a pure include header now, similar to `all.hpp`. 
2. Do we need a changelog entry, as this is a public header? (If yes, please add.)

File: include/seqan3/io/stream/detail/stream_buffer_exposer.hpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+#include <iosfwd>
+
+#include <seqan3/core/platform.hpp>
+
+namespace seqan3::detail
+{
+// ============================================================================
+//  fast_istreambuf_iterator
+// ============================================================================
```
Comment (marehr) [inline include/seqan3/io/stream/detail/stream_buffer_exposer.hpp:24]:
```suggestion
```

[the remark this answers is not in the document]
Comment (marehr) [reply include/seqan3/io/stream/detail/stream_buffer_exposer.hpp]: Done

**Comment by marehr** [comment]
> Thank you! I would like you to answer two questions and make one change:
> 
>     1. Why don't we deprecate the empty file `iterator.hpp`? It is a pure include header now, similar to `all.hpp`.

True, I didn't want to do to much in this PR, but you are right I can remove that old header.

> 
>     2. Do we need a changelog entry, as this is a public header? (If yes, please add.)

This is not a problem, because ALL entities are detail and thus the header.

**Comment by joergi-w** [comment]
Please check the header test...

**Review by joergi-w — APPROVED** [review]

**Review by eseiler — APPROVED** [review]

**Pull request merged**